src/som/interpreter/actors/AbstractPromiseResolutionNode.java

```diff
@@ -5,18 +5,29 @@
 import com.oracle.truffle.api.instrumentation.Instrumentable;
 import com.oracle.truffle.api.source.SourceSection;
 
+import som.VM;
 import som.interpreter.actors.SPromise.Resolution;
 import som.interpreter.actors.SPromise.SResolver;
 import som.interpreter.nodes.nary.TernaryExpressionNode;
+import som.interpreter.nodes.nary.UnaryExpressionNode;
 
 
 @Instrumentable(factory = AbstractPromiseResolutionNodeWrapper.class)
 public abstract class AbstractPromiseResolutionNode extends TernaryExpressionNode {
 
   @Child protected WrapReferenceNode wrapper = WrapReferenceNodeGen.create();
+  @Child protected UnaryExpressionNode haltNode;
 
-  protected AbstractPromiseResolutionNode(final boolean eagWrap, final SourceSection source) { super(eagWrap, source); }
-  protected AbstractPromiseResolutionNode(final AbstractPromiseResolutionNode node) { super(node); }
+  protected AbstractPromiseResolutionNode(final boolean eagWrap, final SourceSection source) {
+    super(eagWrap, source);
+    haltNode = insert(SuspendExecutionNodeGen.create(false, source, 2, null));
+    VM.insertInstrumentationWrapper(haltNode);
```

Comment:
Ehm, I am not questioning the initialization of the `haltNode`.
I am just wondering whether `haltNode = SuspendExecutionNodeGen.create(false, source, 2, null);` would be sufficient.